src/Entity/Form/DeveloperAppCreateFormForDeveloper.php

```diff
@@ -84,52 +73,71 @@ public function __construct(DeveloperAppCredentialControllerFactoryInterface $ap
   public static function create(ContainerInterface $container) {
     return new static(
       $container->get('apigee_edge.controller.developer_app_credential_factory'),
-      $container->get('config.factory'),
-      $container->get('entity.manager'),
-      $container->get('entity_type.manager'),
-      $container->get('module_handler')
+      $container->get('entity_type.manager')
     );
   }
 
   /**
-   * Form constructor.
-   *
-   * @param array $form
-   *   An associative array containing the structure of the form.
-   * @param \Drupal\Core\Form\FormStateInterface $form_state
-   *   The current state of the form.
-   * @param int|null $user
-   *   User id, up-casting is not working, because _entity_form is used instead
-   *   of _form in routing.yml.
-   *
-   * @return array
-   *   The form structure.
+   * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state, UserInterface $user = NULL) {
-    $this->userId = $user->id();
-    $this->checkDeveloperStatus($user->id());
+    // This is the only place where we can grab additional route parameters.
+    // See implementation in parent.
+    $this->user = $user;
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+    $this->checkDeveloperStatus($this->getUser()->id());
 
-    $form = parent::buildForm($form, $form_state);
-    $form['developerId'] = [
+    // The user from the route is the owner.
+    $form['owner'] = [
       '#type' => 'value',
-      '#value' => $this->entity->getDeveloperId(),
+      '#value' => $this->getUser()->get('apigee_edge_developer_id')->value,
     ];
+
     return $form;
   }
 
   /**
    * {@inheritdoc}
    */
-  protected function prepareEntity() {
-    $this->entity->setOwnerId($this->userId);
+  protected function getRedirectUrl(): Url {
+    $entity = $this->getEntity();
+    return $entity->toUrl('collection-by-developer');
+  }
+
+  /**
+   * The user from the route or the current user if it is not available.
+   *
+   * @return \Drupal\user\UserInterface
+   *   User object.
+   */
+  protected function getUser(): UserInterface {
+    return $this->user ?? $this->entityTypeManager->getStorage('user')->load(\Drupal::currentUser()->id());
```

Comment:
Can you inject the `current_user` service?

Reply:
I can, but because this is a fallback solution (that should never be triggered) I did not want to add an extra parameter to the constructor just for this.